### PR TITLE
Update DG: Add implementation details for edelete and eedit features and planned enhancements

### PR DESCRIPTION
**Pull request by tiongMax**
As stated

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -514,24 +514,92 @@ The following sequence diagram shows how the event show operation works:
...
 
-Step 2. When the command is parsed, the index of the event in the event list is then determined by parsing from String to an integer.
+<puml src="diagrams/EventDeleteSequenceDiagram.puml" alt="EventDeleteSequenceDiagram" />
 
-Step 3. The command is then executed and with the index and the event is deleted from the filtered list.
+Step 3. When the `EventDeleteCommand` finishes executing, the updated `EventStorage` is written into `eventStorage.json` 
+file.
 
-Step 4. Then the storage will also be updated accordingly by the filtered list.
+The following the activity diagram for this feature.
```
Comment (brein62) [inline docs/DeveloperGuide.md:540]: minor grammar fix!

`Here is the activity diagram for this feature:`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -514,24 +514,92 @@ The following sequence diagram shows how the event show operation works:
...
+
+* **Alternative 1 (current choice):** Proceed without prefix.
+    * Pros: Easy to implement.
+    * Cons: Lack of consistency.
+
+* **Alternative 2:** Include prefix. 
+    * Pros: Consistency is adhered.
+    * Cons: More steps will be taken to implement and it is redundant since the index is the only field.
+
+### Editing the detail of an event
```
Comment (brein62) [inline docs/DeveloperGuide.md:556]: `detail` -> `details`

**Review by ruo-x — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -514,24 +514,92 @@ The following sequence diagram shows how the event show operation works:
   * Pros: Might be easier to implement as the `EventListPanel` will already have access to the event information as it has `filteredEventList` as one of its fields.
   * Cons: Depletes the user experience as they will have to use the `elist` command, followed by the `eshow` command if they want to see the event information of another event.
 
-### \[In progress\] Delete the event from a list of event
+### Delete the event from a list of events
```
Comment (brein62) [inline docs/DeveloperGuide.md:517]: `Delete the event from a list of events` -> `Delete an event from a list of events`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -514,24 +514,92 @@ The following sequence diagram shows how the event show operation works:
...
 
 ### Implementation
 
-To facilitate the event delete command, the class EventDeleteCommand is created. The class extends from the interface
-Command. When the command is then parsed and executed.
+When deleting an event, the event in the EventStorage will also be deleted. The new event list is then written into the
```
Comment (brein62) [inline docs/DeveloperGuide.md:521]: choose either `EventStorage` (in code form) or event storage

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -514,24 +514,92 @@ The following sequence diagram shows how the event show operation works:
...
+
+Given below is an example usage scenario and how the mechanism of the event edit behaves at each step.
+
+Step 1. The user launches the application and enters the command to edit an event. For example,
+`eedit 1 r/10 cleaner sd/1/1/2023 1100 ed/2/2/2023 2200 l/NUS dsc/Help clean out m/10 clothes b/50.00 vs/10`. When
+iVolunteer receives the input, it will parse the input and split it into command word and the update details which 
+includes the index. The index will then be checked if it is valid by parsing it from String to an integer.
+
+Step 2. If the index is valid, a new `EditEventDescriptor` object will be created. The fields of the 
+`EditEventDescriptor` object will be assigned by the details from user input after they are validated and then passed to
```
Comment (brein62) [inline docs/DeveloperGuide.md:573]: `assigned by` -> `filled with`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -724,6 +792,7 @@ Meanwhile, the activity diagram below shows the general workflow when a `Volunte
 
 _{Explain here how the data archiving feature will be implemented}_
 
+--------------------------------------------------------------------------------------------------------------------
 
 --------------------------------------------------------------------------------------------------------------------
```
Comment (brein62) [inline docs/DeveloperGuide.md:797]: no need for multiple horizontal rules

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1789,6 +1855,12 @@ Currently, the duplicate detection for roles and materials within the Event mode
    - produce an error message `There is more than 1 role/material with the same name: [ROLE/MATERIAL NAME]!` whenever there is a duplicate role/material such that volunteer coordinators are aware of these duplicates and fix the respective `ecreate` or `eedit` command.
    - **Example:** With these changes, the command `ecreate n/Learn farming r/30 farmers r/40 farmers r/100 participants sd/18/11/2023 1230 l/lim chu kang dsc/learn farming` will produce the error message `There is more than 1 role with the same name: farmers!` since `r/30 farmers` and `r/40 farmers` are duplicates.
 
+### Improve uniqueness of the volunteers and events with the same name
+The current implementation of the iVolunteer can only accept volunteer and event with different name.
+However, this is not the best option as new volunteer and event with name existed in the storage must change in order to be
+added into iVolunteer. In order to have volunteer and event with the same name, we plan to implement unique id to
```
Comment (brein62) [inline docs/DeveloperGuide.md:1861]: `id` -> `ID`

**Review by brein62 — CHANGES_REQUESTED** [review]
just a few english suggestions! sorry boss >_<

**Comment by tiongMax** [comment]
> just a few english suggestions! sorry boss >_<

no worries boss, thank you boss

**Pull request merged**